src/mpid/ch4/generic/gpu/mpidig_gpu_recv.c

```diff
@@ -0,0 +1,81 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2020 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+#include "mpidimpl.h"
+
+int MPIDIG_gpu_recv(void *buf,
+                    MPI_Aint count,
+                    MPI_Datatype datatype,
+                    int rank,
+                    int tag,
+                    MPIR_Comm * comm,
+                    int context_offset, MPIDI_av_entry_t * av,
+                    MPI_Status * status, MPIR_Request ** request)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_GPU_RECV);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_GPU_RECV);
+
+    buf = MPIDI_gpu_stage_recv_buf(buf, count, datatype);
```

Comment:
Instead of overwriting `buf`, it is probably better to use a separate local variable.

Reply:
Probably is better to have it return `mpi_errno`.

Reply:
Probably, this was made as it is when I was invoking the function inside the netmod so to avoid changing the buffer name in other places. Now that the call is confined inside the GPU generic code I can change this to also return additional buffer information and store that inside `MPIDIG_gpu_req_t` instead of using a map.